OpenRA.PathfinderTests/PathfinderTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Moq;
+using NUnit.Framework;
+using OpenRA;
+using OpenRA.Mods.Common.Traits;
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace PathfinderTests
+{
+	[TestFixture]
+	public class PathfinderTests
+	{
+		private IMap BuildFakeMap(int mapWidth, int mapHeight)
+		{
+			var map = new Mock<IMap>();
```

Comment:
Mocking frameworks are a blessing when unit testing and isolating dependencies given how easy it is to create stubs and mocks. Not only that, but they are also verifiable and you can check how many times and which call has been performed on a mock. They are quite easy to configure in a very few lines, just look at it in my unit tests, in a few lines everything is set up.

You could go implementing them by yourself, but you may end up trapping yourself in a mocking hell.